src/provider/provider_os_memory.c

```diff
@@ -193,17 +252,19 @@ static umf_result_t translate_params(umf_os_memory_provider_params_t *in_params,
 
     // NUMA config
     int emptyNodeset = in_params->numa_list_len == 0;
-    result = translate_numa_mode(in_params->numa_mode, emptyNodeset,
-                                 &provider->numa_policy);
+    result = validate_numa_mode(in_params->numa_mode, emptyNodeset);
     if (result != UMF_RESULT_SUCCESS) {
         LOG_ERR("incorrect NUMA mode: %u", in_params->numa_mode);
         return result;
     }
 
+    int manual_mode = advancedMembind(in_params);
+    provider->numa_policy =
+        translate_numa_mode(in_params->numa_mode, manual_mode);
     provider->numa_flags = getHwlocMembindFlags(in_params->numa_mode);
-
+    provider->part_size = in_params->part_size;
     return nodemask_to_hwloc_nodeset(
-        in_params->numa_list, in_params->numa_list_len, &provider->nodeset);
+        in_params->numa_list, in_params->numa_list_len, provider, manual_mode);
```

Comment:
so is_separate is really a part_size > 0?

Reply:
For now yes - but when we will add other features, like binding by size this will be more advanced logic - and this is why this is separate variable and function to set it.

Reply:
please choose one from "advanced_membind", "manual_mode", "is_separate" or my proposal "set_separate_nodes" and use it consisitently in the whole code